### ISSUE TITLE
flux full fine-tuning output bad result

### ISSUE THREAD

**Issue by huxian0402** [kohya-ss/sd-scripts#1587]
@lansing @rockerBOO @akx @tsukimiya @wkpark I used the code from https://github.com/kohya-ss/sd-scripts/tree/sd3 branch for FLUX.1 fine-tuning, but the generated images are very blurry. However, using the same data with FLUX.1 LoRA training produces good results. Could it be that there is a bug in the FLUX.1 fine-tuning code? Could you please take a look for me?
FLUX.1 fine-tuning config: 
CUDA_VISIBLE_DEVICES=1,2,3,4,5,6 accelerate launch --config_file base_configs/accelerate_config.yaml  --mixed_precision bf16 --num_cpu_threads_per_process 1 flux_train.py   \
    --pretrained_model_name_or_path flux1-dev.sft   \
    --clip_l clip_l.safetensors    \
    --t5xxl t5xxl_fp16.safetensors    \
    --ae ae.sft  \
    --resolution "1280,768"  \
    --enable_bucket   \
    --min_bucket_reso  256  \
    --max_bucket_reso  1280  \
    --bucket_reso_steps  64  \
    --save_model_as safetensors --sdpa --persistent_data_loader_workers --max_data_loader_n_workers 2   \
    --seed 42 --gradient_checkpointing --mixed_precision bf16 --save_precision bf16   \
    --train_data_dir  ${TRAIN_DATA_DIR}  --output_dir out_weights/${EXP_NAME} --output_name flux-sft   \
    --learning_rate 5e-5 --max_train_epochs 4  --sdpa --highvram --cache_text_encoder_outputs_to_disk --cache_latents_to_disk --save_every_n_epochs 1   \
    --optimizer_type adafactor --optimizer_args "relative_step=False" "scale_parameter=False" "warmup_init=False"   \
    --lr_scheduler constant_with_warmup --max_grad_norm 0.0   \
    --timestep_sampling shift --discrete_flow_shift 3.1582 --model_prediction_type raw --guidance_scale 1.0   \
    --fused_backward_pass --cpu_offload_checkpointing --full_bf16 

FLUX.1 LoRA training config:
 CUDA_VISIBLE_DEVICES=0 accelerate launch --config_file base_configs/default_config.yaml  --mixed_precision bf16 --num_cpu_threads_per_process 1 flux_train_network.py \
    --pretrained_model_name_or_path flux1-dev.sft   \
    --clip_l clip_l.safetensors    \
    --t5xxl t5xxl_fp16.safetensors    \
    --ae ae.sft    \
    --network_dim 4    \
    --learning_rate 1e-4    \
    --train_batch_size 1   \
    --max_train_epochs 8    \
    --save_every_n_epochs 2   \
    --train_data_dir  ${TRAIN_DATA_DIR}    \
    --resolution "1280,768"  \
    --enable_bucket   \
    --min_bucket_reso  256  \
    --max_bucket_reso  1280  \
    --bucket_reso_steps  64  \
    --output_dir out_weights/${EXP_NAME}    \
    --output_name flux-lora   \
    --save_model_as safetensors   \
    --sdpa    \
    --persistent_data_loader_workers    \
    --max_data_loader_n_workers 2    \
    --gradient_checkpointing    \
    --mixed_precision bf16    \
    --save_precision bf16    \
    --network_module networks.lora_flux    \
    --optimizer_type adamw8bit    \
    --fp8_base    \
    --highvram    \
    --timestep_sampling shift    \
    --discrete_flow_shift 3.1582    \
    --model_prediction_type raw    \
    --guidance_scale 1.0  \
    --shuffle_caption  \
    --seed 42

**Comment by huxian0402** [comment]
> The sample learning rate of `5e-5` seems a bit too high in some cases. Could you try a learning rate of around 1e-6 to 5e-6?

Thank you for your response. Following your suggestion, after adjusting the learning rate from 5e-5 to 1e-5, Flux full fine-tuning can output good results similar to LoRA. Great work!

**Comment by Lacrischen** [comment]
Thank you for the valuable issue and the comments. 
I've encountered the same issue and I'm running on the similar setting except the batch size and the training of the text encoder. 
@huxian0402 May I ask you to share the training dataset and the batch size you used? 

Before adapting to my case, I'd like to follow your successful case and learn from it.
Thank you :)

**Comment by Doris-UESTC** [comment]
> > The sample learning rate of `5e-5` seems a bit too high in some cases. Could you try a learning rate of around 1e-6 to 5e-6?
> 
> Thank you for your response. Following your suggestion, after adjusting the learning rate from 5e-5 to 1e-5, Flux full fine-tuning can output good results similar to LoRA. Great work!

Hello, could you tell me the number of images used for LoRA training and full fine-tuning, respectively? I use same setting to train model but it output bad results. Thank you!
